components/nimbus/android/src/main/java/org/mozilla/experiments/nimbus/NimbusBuilder.kt

```diff
@@ -0,0 +1,144 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.experiments.nimbus
+
+import android.content.Context
+import android.net.Uri
+import androidx.annotation.RawRes
+import kotlinx.coroutines.runBlocking
+import java.util.Locale
+
+private const val TIME_OUT_LOADING_EXPERIMENT_FROM_DISK_MS = 200L
+
+/**
+ * A builder for [Nimbus] singleton objects, parameterized in a declarative class.
+ */
+abstract class AbstractNimbusBuilder<T : NimbusInterface>(val context: Context) {
+
+    /**
+     * An optional server URL string.
+     *
+     * This will only be null or empty in development or testing, or in any build variant of a
+     * non-Mozilla fork.
+     */
+    var url: String? = null
+
+    /**
+     * A closure for reporting errors from Rust.
+     */
+    var errorReporter: ErrorReporter = { _: String, _: Throwable -> }
+
+    /**
+     * A flag to select the main or preview collection of remote settings. Defaults to `false`.
+     */
+    var usePreviewCollection: Boolean = false
+
+    /**
+     * A flag to indicate if this is being run on the first run of the app. This is used to control
+     * whether the `initial_experiments` file is used to populate Nimbus.
+     */
+    var isFirstRun: Boolean = true
+
+    /**
+     * A optional raw resource of a file downloaded at or near build time from Remote Settings.
+     */
+    @RawRes
+    var initialExperiments: Int? = null
```

Comment:
I wonder if converting this to be a String type would be helpful as we plan to integrate Nimbus into other apps. We could have other Kotlin apps that aren't Android, and as far as I'm aware the raw resource int thing is an Android thing.

If it's the string value of the initial experiments, I think that also lends itself a bit better to testing and extensibility.

Reply:
I think at this point, building out a non-Android Kotlin version would be a quite a large undertaking (for example, we depend on `android.content.Context`), for currently not much demand.

I think I'm inclined to say "Let's cross that bridge when we get to it".

Reply:
Sounds good to me!